apis/iam/v1beta1/partialpolicy_types.go

```diff
@@ -30,7 +30,6 @@ type MemberReference struct {
 }
 
 // MemberSource represents a source for an IAM identity
-// +kubebuilder:validation:XValidation:rule="(has(self.bigQueryConnectionConnectionRef) ? 1 : 0) + (has(self.logSinkRef) ? 1 : 0) + (has(self.serviceAccountRef) ? 1 : 0) + (has(self.serviceIdentityRef) ? 1 : 0) + (has(self.sqlInstanceRef) ? 1 : 0) == 1",message="exactly one of bigQueryConnectionConnectionRef, logSinkRef, serviceAccountRef, serviceIdentityRef, or sqlInstanceRef must be specified"
```

Comment:
Do you happen to know the context about these validation rules? Or do we plan to introduce the oneOf rules back instead of using CEL?

Reply:
( I think this commit in particular is out of date FYI)

I tried to use CEL rules here but gave up as we'd need the introduce a maxLen for the array field. I chose to go back to the CRD patching. We can explore some more about CEL/ adding maxLen but would like to do that outside of this PR.